pkg/liqo-controller-manager/resource-request-controller/local-resource-monitor.go

```diff
@@ -69,69 +63,45 @@ const (
-	nodesFactory := informers.NewSharedInformerFactoryWithOptions(clientset, resyncPeriod, informers.WithTweakListOptions(virtualNodeFilter))
-	b.nodeInformer = nodesFactory.Core().V1().Nodes().Informer()
-	b.nodeInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
-		AddFunc:    b.onNodeAdd,
-		UpdateFunc: b.onNodeUpdate,
-		DeleteFunc: b.onNodeDelete,
+// NewLocalMonitor creates a new LocalResourceMonitor.
+func NewLocalMonitor(ctx context.Context, clientset kubernetes.Interface, resyncPeriod time.Duration,
+	updater *OfferUpdater) *LocalResourceMonitor {
+	nodeInformer := informers.NewSharedInformerFactoryWithOptions(
+		clientset, resyncPeriod, informers.WithTweakListOptions(noVirtualNodesFilter),
+	).Core().V1().Nodes().Informer()
+	podInformer := informers.NewSharedInformerFactoryWithOptions(
+		clientset, resyncPeriod, informers.WithTweakListOptions(noShadowPodsFilter),
+	).Core().V1().Pods().Informer()
+
+	accountant := LocalResourceMonitor{
+		allocatable:    corev1.ResourceList{},
+		resourcePodMap: map[string]corev1.ResourceList{},
+		nodeMutex:      sync.RWMutex{},
+		podMutex:       sync.RWMutex{},
+		nodeInformer:   nodeInformer,
+		podInformer:    podInformer,
+		updater:        updater,
+	}
+
+	nodeInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    accountant.onNodeAdd,
+		UpdateFunc: accountant.onNodeUpdate,
+		DeleteFunc: accountant.onNodeDelete,
 	})
-	podsFactory := informers.NewSharedInformerFactoryWithOptions(clientset, resyncPeriod, informers.WithTweakListOptions(shadowPodFilter))
-	b.podInformer = podsFactory.Core().V1().Pods().Informer()
-	b.podInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
-		AddFunc:    b.onPodAdd,
-		UpdateFunc: b.onPodUpdate,
-		DeleteFunc: b.onPodDelete,
+	podInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    accountant.onPodAdd,
+		UpdateFunc: accountant.onPodUpdate,
+		DeleteFunc: accountant.onPodDelete,
 	})
 
-	return nil
-}
-
-// StartBroadcaster starts two shared Informers, one for nodes and one for pods launching two separated goroutines.
-func (b *Broadcaster) StartBroadcaster(ctx context.Context, group *sync.WaitGroup) {
-	group.Add(3)
-	go b.updater.Start(ctx, group)
-	go b.startNodeInformer(ctx, group)
-	go b.startPodInformer(ctx, group)
-}
-
-func (b *Broadcaster) startNodeInformer(ctx context.Context, group *sync.WaitGroup) {
-	defer group.Done()
-	b.nodeInformer.Run(ctx.Done())
-}
-
-func (b *Broadcaster) startPodInformer(ctx context.Context, group *sync.WaitGroup) {
-	defer group.Done()
-	b.podInformer.Run(ctx.Done())
-}
-
-func (b *Broadcaster) getPodMap() map[string]corev1.ResourceList {
-	b.podMutex.RLock()
-	defer b.podMutex.RUnlock()
-	mapCopy := make(map[string]corev1.ResourceList)
-	for key, value := range b.resourcePodMap {
-		mapCopy[key] = value.DeepCopy()
-	}
-	return mapCopy
-}
+	go accountant.nodeInformer.Run(ctx.Done())
+	go accountant.podInformer.Run(ctx.Done())
```

Comment:
Typically, informers are started through the factory, and not directly. Additionally, it is usually appropriate to wait for the caches to sync before considering a component to be started.